src/Immediate.Handlers.Shared/IHandler.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace Immediate.Handlers.Shared;
+
+///  <summary>
+///  Represents an abstraction implemented by the source-generated handlers
+///  </summary>
+///  <typeparam name="TRequest">
+///    The type of the command request 
+///  </typeparam>
+///  <typeparam name="TResponse">
+///    The type of the command response
+/// </typeparam>
+public interface IHandler<in TRequest, TResponse>
```

Comment:
should response not be `out TResponse`?

alternate question: why `in` in the first place? handlers shouldn't be passed around, so there's no reason to worry about covariance, i don't think?

Reply:
I think _out_ wouldn't work, since ValueTask<T> in invariant again. But yeah, let's just skip the variance, I agree.